store/card/card.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2019 Drone.IO Inc. All rights reserved.
+// Use of this source code is governed by the Drone Non-Commercial License
+// that can be found in the LICENSE file.
+// +build !oss
+
+package card
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+
+	"github.com/drone/drone/core"
+	"github.com/drone/drone/store/shared/db"
+)
+
+// New returns a new card database store.
+func New(db *db.DB) core.CardStore {
+	return &cardStore{
+		db: db,
+	}
+}
+
+type cardStore struct {
+	db *db.DB
+}
+
+func (c *cardStore) FindCardByBuild(ctx context.Context, build int64) ([]*core.Card, error) {
+	var out []*core.Card
+	err := c.db.View(func(queryer db.Queryer, binder db.Binder) error {
+		params := map[string]interface{}{
+			"card_build": build,
+		}
+		stmt, args, err := binder.BindNamed(queryByBuild, params)
+		if err != nil {
+			return err
+		}
+		rows, err := queryer.Query(stmt, args...)
+		if err != nil {
+			return err
+		}
+		out, err = scanRows(rows)
+		return err
+	})
+	return out, err
+}
+
+func (c cardStore) FindCard(ctx context.Context, step int64) (*core.Card, error) {
+	out := &core.Card{Step: step}
+	err := c.db.View(func(queryer db.Queryer, binder db.Binder) error {
+		params, err := toParams(out)
+		if err != nil {
+			return err
+		}
+		query, args, err := binder.BindNamed(queryByStep, params)
+		if err != nil {
+			return err
+		}
+		row := queryer.QueryRow(query, args...)
+		return scanRow(row, out)
+	})
+	return out, err
+}
+
+func (c cardStore) FindCardData(ctx context.Context, id int64) (io.ReadCloser, error) {
+	out := &core.CardData{}
+	err := c.db.View(func(queryer db.Queryer, binder db.Binder) error {
+		params := map[string]interface{}{
+			"card_id": id,
+		}
+		query, args, err := binder.BindNamed(queryKey, params)
+		if err != nil {
+			return err
+		}
+		row := queryer.QueryRow(query, args...)
+		return scanRowCardDataOnly(row, out)
+	})
+	return ioutil.NopCloser(
+		bytes.NewBuffer(out.Data),
+	), err
+}
+
+func (c cardStore) CreateCard(ctx context.Context, card *core.CreateCard) error {
```

Comment:
the `data` should be removed from the `Card` and instead passed in as an `io.Reader` so that we can (eventually) allow uploading and storing in S3
https://gist.github.com/bradrydzewski/0b8b67db06f9fcf8dcb718c6fdf8e658#database-calls

Reply:
done